### PR TITLE
Add embedded material editor to OceanRenderer

### PR DESCRIPTION
**Pull request by daleeidd**
Adds a material editor similar to what the mesh renderers get. The only difference is that this appears right after the component rather than after all other components.

I wasn't able to make the feature an attribute like the SO editor. This is more difficult as the foldout should appear after the component at the very least. It is probably possible but will require more engineering than the other.

### CODE REVIEW

File: crest/Assets/Crest/Crest/Scripts/OceanRenderer.cs [hunk 1]
```diff
@@ -1324,6 +1347,45 @@ public override void OnInspectorGUI()
...
+                {
+                    // Free the memory used by the previous MaterialEditor.
+                    DestroyImmediate(_materialEditor);
+                }
+
+                if (_target._material != null)
+                {
+                    // Create a new instance of the default MaterialEditor.
+                    _materialEditor = (MaterialEditor)CreateEditor(_target._material);
+
```
Comment (huwb) [inline crest/Assets/Crest/Crest/Scripts/OceanRenderer.cs:1371]: whitespace can be removed

**Comment by huwb** [comment]
Wow nice! I think this is fine as is.

I dont know if we ever supported this but i tried clearing the material reference, got one null ref exception which i pushed a commit to fix, then it seems to null ref except in the new code, in a slightly confusing way and i wasnt certain the best fix. Would it be possible to support clearing the material off the component more gracefully?

**Comment by daleeidd** [comment]
Thanks. All fixed.

> I dont know if we ever supported this but i tried clearing the material reference, got one null ref exception which i pushed a commit to fix, then it seems to null ref except in the new code, in a slightly confusing way and i wasnt certain the best fix. Would it be possible to support clearing the material off the component more gracefully?

That was an awful conditional statement. I parted it out to solve the issue.

**Review by huwb — APPROVED** [review]
This is super awesome

**Comment by daleeidd** [comment]
Thanks. Merged :^)

**Pull request merged**